x-pack/filebeat/input/httpjson/config_request.go

```diff
@@ -12,6 +12,8 @@ import (
 	"strings"
 	"time"
 
+	"gopkg.in/natefinch/lumberjack.v2"
```

Comment:
We already have a logging library in Beats: https://github.com/elastic/elastic-agent-libs/tree/main/logp that uses [go.uber.org/zap](https://pkg.go.dev/go.uber.org/zap) under the hood.

If possible I'd prefer to stick with those ones (preferably `elastic-agent-libs/logp`) to avoid introducing a new dependency and "logging" library.

What are the advantages of using lumberjack over the ones we already have in Beats?

Reply:
Another alternative is https://pkg.go.dev/github.com/elastic/elastic-agent-libs/file#Rotator. It gives you an `io.WriteCloser` interface and manages rolling the underlying files.

But it looks like the library got polluted since I last looked with a hardcoded file extension 😢 . https://github.com/elastic/elastic-agent-libs/blob/a09d65fd12dec3efe53c44f05562431f96b1028b/file/rotator.go#L416  IMO is should have used `filepath.Ext` to get the extension from the configured `filename`.

Reply:
> What are the advantages of using lumberjack over the ones we already have in Beats?

Simplicity; this does exactly what is needed and is transparently simple.

Reply:
@andrewkroh's suggestion looks great and simple. The file extension is a pitty, but we can easily fix that as well. I glanced over the code and it seems to be possible to make the extension configurable without changing the public API.